python_modules/dagster/dagster/core/telemetry.py

```diff
@@ -275,11 +275,17 @@ def _check_telemetry_instance_param(args, kwargs, instance_index):
 
 
 def _get_telemetry_logger():
+
+    # If a concurrently running process deleted the logging directory since the
+    # last action, we need to make sure to re-create the directory
+    # (the logger does not do this itself.)
+    get_or_create_dir_from_dagster_home("logs")
+
     logger = logging.getLogger("dagster_telemetry_logger")
 
     if len(logger.handlers) == 0:
         handler = RotatingFileHandler(
-            os.path.join(get_dir_from_dagster_home("logs"), "event.log"),
+            os.path.join(get_or_create_dir_from_dagster_home("logs"), "event.log"),
             maxBytes=MAX_BYTES,
             backupCount=10,
         )
```

Comment:
<!-- GRAPHITE_METADATA {"trueLocation":{"startLine":277,"endLine":295}} -->
<i class='graphite__hidden'>[Re: lines 277 to 295]<br/><br/></i>

looks like this fn is only called once - might be cleaner to merge it . It was not obvious to me that `_get_telemtry_logger` was called every invocation instead of just once at the beginning.

Reply:
Hm I personally like that the logic of initializing the logger is isolated to its own function, I find it a bit easier to reason about.

Reply:
>  not obvious to me that _get_telemtry_logger was called every invocation instead of just once at the beginning.

as long as we have something to mitigate this concern from happening in a refactor, since we are depending on that behavior to ensure the directory is created every time and there are no tests protecting it currently

You could write a test that called it - removed the directory - and called it again. That would guard.